portal/src/main/webapp/js/src/study-view/data/StudyViewProxy.js

```diff
@@ -412,6 +413,117 @@ var StudyViewProxy = (function() {
+                        gene: data.gene[i],
+                        gistic: data.gistic[i]
+                    });
+                }
+            }
+        }
+        return converted;
+    }
+
+    function getCNABasedOnSampleIds(sampleIds) {
+        var data = {
+            alter: [],
+            caseIds: [],
+            cytoband: [],
+            gistic: [],
+            gene: []
+        };
+        if (sampleIds instanceof Array) {
+            if (sampleIds.length === this.getSampleIds().length) {
+                return obtainDataObject.cna;
+            }
+            if (obtainDataObject.hasOwnProperty('cnaSampleBased')) {
+                var geneSpecific = {};
+                var numOfSample = sampleIds.length;
+                for (var i = 0; i < numOfSample; i++) {
+                    if (obtainDataObject.cnaSampleBased.hasOwnProperty(sampleIds[i])) {
+                        for (var j = 0, numOfGenes = obtainDataObject.cnaSampleBased[sampleIds[i]].length; j < numOfGenes; j++) {
+                            var key = obtainDataObject.cnaSampleBased[sampleIds[i]][j].gene + obtainDataObject.cnaSampleBased[sampleIds[i]][j].alter;
+                            if (!geneSpecific.hasOwnProperty(key)) {
+                                geneSpecific[key] = obtainDataObject.cnaSampleBased[sampleIds[i]][j];
+                                geneSpecific[key].caseIds = [];
+                            }
+                            geneSpecific[key].caseIds.push(sampleIds[i]);
+                        }
+                    }
+                }
+                _.values(geneSpecific).map(function (item) {
+                    data.alter.push(item.alter);
+                    data.caseIds.push(item.caseIds);
+                    data.cytoband.push(item.cytoband);
+                    data.gistic.push(item.gistic);
+                    data.gene.push(item.gene);
+                });
+            }
+        }
+
+        return data;
+    }
+
+    //Change the mutated gene data base on sample ID
+    function convertMutatedGeneData(data) {
+        var converted = {};
+
+        if (data) {
+            for (var i = 0, dataL = data.length; i < dataL; i++) {
+                var caseIds = data[i].caseIds;
+
+                for (var j = 0, caseIdsL = caseIds.length; j < caseIdsL; j++) {
+                    if (!converted.hasOwnProperty(caseIds[j])) {
+                        converted[caseIds[j]] = [];
+                    }
+                    converted[caseIds[j]].push(data[i]);
+                }
+            }
+        }
+        return converted;
+    }
+
+    function getMutatedGeneDataBasedOnSampleIds(sampleIds) {
+        var data = [];
+        if (sampleIds instanceof Array) {
+            if(sampleIds.length === this.getSampleIds().length) {
+                return obtainDataObject.mutatedGenes;
+            }
+            if(obtainDataObject.hasOwnProperty('mutatedGenesSampleBased')){
```

Comment:
I think it is not such a good practice to use globals like is done here with obtainDataObject. A *small* improvement to this would be to wrap access to this object via a getter. Something like: 
```javascript
function getDataObject() {
   return obtainDataObject;
}
```
still not pretty....but is an improvement. .